spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -89,16 +88,18 @@ protected void handleMessageInternal(Message<?> message) throws Exception {
 
 		ByteString pubsubPayload;
 
-		if (payload instanceof byte[]) {
+		if (this.messageConverter != null) {
+			pubsubPayload = ByteString.copyFrom((byte[]) this.messageConverter.fromMessage(message,
+					byte[].class));
+		}
+		else if (payload instanceof byte[]) {
 			pubsubPayload = ByteString.copyFrom((byte[]) payload);
 		}
 		else if (payload instanceof ByteString) {
 			pubsubPayload = (ByteString) payload;
 		}
 		else {
-			pubsubPayload =	ByteString.copyFrom(
-					(String) this.messageConverter.fromMessage(message, String.class),
-					Charset.defaultCharset());
+			pubsubPayload = ByteString.copyFrom(message.getPayload().toString().getBytes());
```

Comment:
Don't you already have the `payload` variable that you're checking the instanceof  ? `Object payload = message.getPayload();`